Init android w3m docs (#686)
WalletConnect/walletconnect-docs: docs/android/web3modal/usage.md

# Usage

## Web3Modal SDK

> Web3Modal SDK allows easy integration of [Web3Modal](https://web3modal.com/) in a native Android application.

Reviewer: ```suggestion
The Web3Modal Android SDK allows for easy integration of [Web3Modal](https://web3modal.com/) in a native Android application.
```

## Web3Modal Client

> `Web3Modal` is object that interacts with the Web3Modal SDK.

Reviewer: ```suggestion
`Web3Modal` is a singleton that interacts with the Web3Modal SDK.
```

### Initialize

```kotlin
val connectionType = ConnectionType.AUTOMATIC or ConnectionType.MANUAL

> val projectId = PROJECT_ID

Reviewer 2: ```suggestion
val projectId = "" // Get Project ID at https://cloud.walletconnect.com/
```

val relayUrl = "relay.walletconnect.com"
val serverUrl = "wss://$relayUrl?projectId=${projectId}"
val appMetaData = Core.Model.AppMetaData(
    name = "Kotlin.Web3Modal",
    description = "Kotlin Web3Modal Implementation",
    url = "kotlin.web3Modal.walletconnect.com",
    icons = listOf("https://raw.githubusercontent.com/WalletConnect/walletconnect-assets/master/Icon/Gradient/Icon.png"),

>     redirect = null

Reviewer: we should define a redirect to show how the redirect feature works

)

> CoreClient.initialize(relayServerUrl = serverUrl, connectionType = ConnectionType.AUTOMATIC, application = this, metaData = appMetaData)

Reviewer: ```suggestion
CoreClient.initialize(relayServerUrl = serverUrl, connectionType = connectionType, application = this, metaData = appMetaData)
```

Web3Modal.initialize(
    init = Modal.Params.Init(CoreClient),
    onSuccess = {
        // Callback will be called if initialization is successful
     },
    onError = { error -> 
        // Error will be thrown if there's an issue during initialization
    }
)
```

## Web3Modal Usage

### Android Compose

````kotlin
    setContent {
        val modalSheetState = rememberModalBottomSheetState(initialValue = ModalBottomSheetValue.Hidden, skipHalfExpanded = true)
        val bottomSheetNavigator = BottomSheetNavigator(modalSheetState)

>         val navController = rememberNavController(bottomSheetNavigator)

Reviewer: We should also add the compose libraries to the list of required dependencies that are needed to use W3M

>         val navController = rememberNavController(bottomSheetNavigator)

Author: Added

        ModalBottomSheetLayout(bottomSheetNavigator = bottomSheetNavigator) {
            NavHost(
                navController = navController,
                startDestination = "home"
            ) {
                composable("home") {
                    HomeScreen()
                }
                web3ModalGraph(navController, modalSheetState)
            }
        }
    }
````

> **IMPORTANT**: Web3modal uses accompanist navigation material inside. `ModalBottomSheetLayout` should be imported from this package

Reviewer 2: Which package do you mean here? Could you add a link to `this`?

Author: Link to accompanist has been added

        // Triggered when receives the session rejection from wallet
    }

    override fun onSessionUpdate(updatedSession: Web3Modal.Model.UpdatedSession) {
        // Triggered when receives the session update from wallet
    }

    override fun onSessionExtend(session: Web3Modal.Model.Session) {
        // Triggered when receives the session extend from wallet
    }

    override fun onSessionEvent(sessionEvent: Web3Modal.Model.SessionEvent) {
        // Triggered when the peer emits events that match the list of events agreed upon session settlement
    }

    override fun onSessionDelete(deletedSession: Web3Modal.Model.DeletedSession) {
        // Triggered when receives the session delete from wallet
    }

    override fun onSessionRequestResponse(response: Web3Modal.Model.SessionRequestResponse) {
        // Triggered when receives the session request response from wallet
    }

    override fun onConnectionStateChange(state: Web3Modal.Model.ConnectionState) {
        //Triggered whenever the connection state is changed
    }

    override fun onError(error: Web3Modal.Model.Error) {
        // Triggered whenever there is an issue inside the SDK
    }
}
````

> The Web3Modal needs a `Web3Modal.ModalDelegate` passed to it for it to be able to expose asynchronously updates sent from the Wallet.

Reviewer 2: Where do you call `setDelegate()`?

val chains: List<String> = /*List of chains that wallet will be requested for*/
val methods: List<String> = /*List of methods that wallet will be requested for*/
val events: List<String> = /*List of events that wallet will be requested for*/
val requiredNamespaces: Map<String, Web3Modal.Model.Namespaces.Proposal> = mapOf(namespace, Web3Modal.Model.Namespaces.Proposal(accounts, methods, events)) /*Required namespaces to setup a session*/
val optionalNamespaces: Map<String, Web3Modal.Model.Namespaces.Proposal> = mapOf(namespace, Web3Modal.Model.Namespaces.Proposal(accounts, methods, events)) /*Optional namespaces to setup a session*/
val pairing: Core.Model.Pairing = /*Either an active or inactive pairing*/
val connectParams = Web3Modal.Params.Connect(requiredNamespaces, optionalNamespaces, pairing)

Web3Modal.connect(connectParams, 
    { onSuccess -> 
        /* callback that letting you know that you have successfully initiated connecting */ 
    }, 
    { error -> 
        /* callback for error while trying to initiate a connection with a peer */ 

>     }

Reviewer: let's use named arguments here

)
````

More about optional and required namespaces can be found [here](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md)

#
## Disconnect

````kotlin
val disconnectParams = Web3Modal.Params.Disconnect(topic)

Web3Modal.disconnect(
    disconnectParams,
    { 
        /* callback that letting you know that you have successfully disconnected */ 
    },
    { error -> 
        /* callback for error while trying to disconnection with a peer */ 

>     }

Reviewer: let's use named arguments here

)
````

#
## Request
````kotlin
val requestParams = Modal.Params.Request(
    sessionTopic = sessionTopic,
    method = /* Selected method */,
    params = /* Method params  */,
    chainId = /* Chain id */
)

Web3Modal.request(requestParams,
    {
        /* callback that letting you know that you have successful request */
    },
    { error ->
        /* callback for error */

>     }

Reviewer: let's use named arguments here
